### PR TITLE
1005  feature/user role matrix

### PR DESCRIPTION
**Pull request by greg0ire**
This one was way harder than the previous one! Please tell me what you think

### CODE REVIEW

File: src/Security/RolesBuilder/SecurityRolesBuilder.php [hunk 1]
```diff
@@ -54,48 +54,69 @@ public function __construct(
...
+            $translatedRoles = array_map(
+                [$this, 'translateRole'],
+                $childRoles,
+                array_fill(0, count($childRoles), $domain)
+            );
+
             $securityRoles[$role] = [
                 'role' => $role,
-                'role_translated' => $this->translateRole($role, $domain),
+                'role_translated' => $role.': '.implode(', ', $translatedRoles),
```
Comment (greg0ire) [inline src/Security/RolesBuilder/SecurityRolesBuilder.php:69]: Kept the same trailing `: `, not sure what to do…

**Comment by greg0ire** [comment]
@silasjoisten @kunicmarko20 @jordisala1991 please review

**Review by silasjoisten — APPROVED** [review]

File: src/Security/RolesBuilder/SecurityRolesBuilder.php (hunk 1 above)
Comment (kunicmarko20) [reply src/Security/RolesBuilder/SecurityRolesBuilder.php:69]: check before adding `:`?

```
$translatedRoles = count($translatedRoles) > 0 ? ': '.implode(', ', $translatedRoles) : '';
$securityRoles[$role] = [
    'role' => $role,
    'role_translated' => $role . $translatedRoles
    'is_granted' => $this->authorizationChecker->isGranted($role),
];
```

**Comment by kunicmarko20** [comment]
It looks ok, just that `:` thing.

**Comment by greg0ire** [comment]
Then if you're ok with the naming and the numerous interfaces, please merge @silasjoisten , and then either you or I will fix that dangling colon. I might be able to do this tonight, but late.

**Review by kunicmarko20 — APPROVED** [review]

**Pull request merged**